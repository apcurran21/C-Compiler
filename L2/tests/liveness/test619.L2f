(@main
	0
	%arg1 <- 10
	%invalid <- %arg1 < 2
	cjump %invalid = 1 :call_label2
	%arg1 <<= 1
	%arg1 += 1
	rdi <- %arg1
	rsi <- 3
	call allocate 2
	%arrayAddr <- rax
	%pos0 <- %arrayAddr
	%pos0 += 8
	%pos1 <- %arrayAddr
	%pos1 += 16
	%pos2 <- %arrayAddr
	%pos2 += 24
	mem %pos0 0 <- 0
	mem %pos1 0 <- 1
	%arrayLen <- mem %arrayAddr 0
	%count <- 3
	:call_label0
	%ifDone <- %arrayLen < %count
	cjump %ifDone = 1 :call_label1
	%val0 <- mem %pos0 0
	%val1 <- mem %pos1 0
	%val2 <- %val0
	%val2 += %val1
	mem %pos2 0 <- %val2
	%count += 1
	%pos0 += 8
	%pos1 += 8
	%pos2 += 8
	goto :call_label0
	:call_label1
	rdi <- %arrayAddr
	mem rsp -8 <- :call_label3
	call @array_encode 1
	:call_label3
	rdi <- %arrayAddr
	call print 1
	:call_label2
	return
)
