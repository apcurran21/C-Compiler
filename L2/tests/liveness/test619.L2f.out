(
(in
(r12 r13 r14 r15 rax rbp rbx)
(%arg1 r12 r13 r14 r15 rax rbp rbx)
(%arg1 %invalid r12 r13 r14 r15 rax rbp rbx)
(%arg1 r12 r13 r14 r15 rbp rbx)
(%arg1 r12 r13 r14 r15 rbp rbx)
(%arg1 r12 r13 r14 r15 rbp rbx)
(r12 r13 r14 r15 rbp rbx rdi)
(r12 r13 r14 r15 rbp rbx rdi rsi)
(r12 r13 r14 r15 rax rbp rbx)
(%arrayAddr r12 r13 r14 r15 rbp rbx)
(%arrayAddr %pos0 r12 r13 r14 r15 rbp rbx)
(%arrayAddr %pos0 r12 r13 r14 r15 rbp rbx)
(%arrayAddr %pos0 %pos1 r12 r13 r14 r15 rbp rbx)
(%arrayAddr %pos0 %pos1 r12 r13 r14 r15 rbp rbx)
(%arrayAddr %pos0 %pos1 %pos2 r12 r13 r14 r15 rbp rbx)
(%arrayAddr %pos0 %pos1 %pos2 r12 r13 r14 r15 rbp rbx)
(%arrayAddr %pos0 %pos1 %pos2 r12 r13 r14 r15 rbp rbx)
(%arrayAddr %pos0 %pos1 %pos2 r12 r13 r14 r15 rbp rbx)
(%arrayAddr %arrayLen %pos0 %pos1 %pos2 r12 r13 r14 r15 rbp rbx)
(%arrayAddr %arrayLen %count %pos0 %pos1 %pos2 r12 r13 r14 r15 rbp rbx)
(%arrayAddr %arrayLen %count %pos0 %pos1 %pos2 r12 r13 r14 r15 rbp rbx)
(%arrayAddr %arrayLen %count %ifDone %pos0 %pos1 %pos2 r12 r13 r14 r15 rbp rbx)
(%arrayAddr %arrayLen %count %pos0 %pos1 %pos2 r12 r13 r14 r15 rbp rbx)
(%arrayAddr %arrayLen %count %pos0 %pos1 %pos2 r12 r13 r14 r15 rbp rbx %val0)
(%arrayAddr %arrayLen %count %pos0 %pos1 %pos2 r12 r13 r14 r15 rbp rbx %val0 %val1)
(%arrayAddr %arrayLen %count %pos0 %pos1 %pos2 r12 r13 r14 r15 rbp rbx %val1 %val2)
(%arrayAddr %arrayLen %count %pos0 %pos1 %pos2 r12 r13 r14 r15 rbp rbx %val2)
(%arrayAddr %arrayLen %count %pos0 %pos1 %pos2 r12 r13 r14 r15 rbp rbx)
(%arrayAddr %arrayLen %count %pos0 %pos1 %pos2 r12 r13 r14 r15 rbp rbx)
(%arrayAddr %arrayLen %count %pos0 %pos1 %pos2 r12 r13 r14 r15 rbp rbx)
(%arrayAddr %arrayLen %count %pos0 %pos1 %pos2 r12 r13 r14 r15 rbp rbx)
(%arrayAddr %arrayLen %count %pos0 %pos1 %pos2 r12 r13 r14 r15 rbp rbx)
(%arrayAddr r12 r13 r14 r15 rbp rbx)
(%arrayAddr r12 r13 r14 r15 rbp rbx)
(%arrayAddr r12 r13 r14 r15 rbp rbx rdi)
(%arrayAddr r12 r13 r14 r15 rbp rbx rdi)
(%arrayAddr r12 r13 r14 r15 rbp rbx)
(%arrayAddr r12 r13 r14 r15 rbp rbx)
(r12 r13 r14 r15 rbp rbx rdi)
(r12 r13 r14 r15 rax rbp rbx)
(r12 r13 r14 r15 rax rbp rbx)
)

(out
(%arg1 r12 r13 r14 r15 rax rbp rbx)
(%arg1 %invalid r12 r13 r14 r15 rax rbp rbx)
(%arg1 r12 r13 r14 r15 rax rbp rbx)
(%arg1 r12 r13 r14 r15 rbp rbx)
(%arg1 r12 r13 r14 r15 rbp rbx)
(r12 r13 r14 r15 rbp rbx rdi)
(r12 r13 r14 r15 rbp rbx rdi rsi)
(r12 r13 r14 r15 rax rbp rbx)
(%arrayAddr r12 r13 r14 r15 rbp rbx)
(%arrayAddr %pos0 r12 r13 r14 r15 rbp rbx)
(%arrayAddr %pos0 r12 r13 r14 r15 rbp rbx)
(%arrayAddr %pos0 %pos1 r12 r13 r14 r15 rbp rbx)
(%arrayAddr %pos0 %pos1 r12 r13 r14 r15 rbp rbx)
(%arrayAddr %pos0 %pos1 %pos2 r12 r13 r14 r15 rbp rbx)
(%arrayAddr %pos0 %pos1 %pos2 r12 r13 r14 r15 rbp rbx)
(%arrayAddr %pos0 %pos1 %pos2 r12 r13 r14 r15 rbp rbx)
(%arrayAddr %pos0 %pos1 %pos2 r12 r13 r14 r15 rbp rbx)
(%arrayAddr %arrayLen %pos0 %pos1 %pos2 r12 r13 r14 r15 rbp rbx)
(%arrayAddr %arrayLen %count %pos0 %pos1 %pos2 r12 r13 r14 r15 rbp rbx)
(%arrayAddr %arrayLen %count %pos0 %pos1 %pos2 r12 r13 r14 r15 rbp rbx)
(%arrayAddr %arrayLen %count %ifDone %pos0 %pos1 %pos2 r12 r13 r14 r15 rbp rbx)
(%arrayAddr %arrayLen %count %pos0 %pos1 %pos2 r12 r13 r14 r15 rbp rbx)
(%arrayAddr %arrayLen %count %pos0 %pos1 %pos2 r12 r13 r14 r15 rbp rbx %val0)
(%arrayAddr %arrayLen %count %pos0 %pos1 %pos2 r12 r13 r14 r15 rbp rbx %val0 %val1)
(%arrayAddr %arrayLen %count %pos0 %pos1 %pos2 r12 r13 r14 r15 rbp rbx %val1 %val2)
(%arrayAddr %arrayLen %count %pos0 %pos1 %pos2 r12 r13 r14 r15 rbp rbx %val2)
(%arrayAddr %arrayLen %count %pos0 %pos1 %pos2 r12 r13 r14 r15 rbp rbx)
(%arrayAddr %arrayLen %count %pos0 %pos1 %pos2 r12 r13 r14 r15 rbp rbx)
(%arrayAddr %arrayLen %count %pos0 %pos1 %pos2 r12 r13 r14 r15 rbp rbx)
(%arrayAddr %arrayLen %count %pos0 %pos1 %pos2 r12 r13 r14 r15 rbp rbx)
(%arrayAddr %arrayLen %count %pos0 %pos1 %pos2 r12 r13 r14 r15 rbp rbx)
(%arrayAddr %arrayLen %count %pos0 %pos1 %pos2 r12 r13 r14 r15 rbp rbx)
(%arrayAddr r12 r13 r14 r15 rbp rbx)
(%arrayAddr r12 r13 r14 r15 rbp rbx rdi)
(%arrayAddr r12 r13 r14 r15 rbp rbx rdi)
(%arrayAddr r12 r13 r14 r15 rbp rbx)
(%arrayAddr r12 r13 r14 r15 rbp rbx)
(r12 r13 r14 r15 rbp rbx rdi)
(r12 r13 r14 r15 rax rbp rbx)
(r12 r13 r14 r15 rax rbp rbx)
()
)

)
